NETProvider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbParameterCollection.cs

```diff
@@ -90,6 +91,21 @@ public override object SyncRoot
 			get { return ((ICollection)_parameters).SyncRoot; }
 		}
 
+		internal bool? _collectionHasParameterWithUnicodeName;
```

Comment:
I don't want to recompute field on each change. I intentionally made computing the field only on-demand if ```IndexOf``` is called.